gitpod-io/gitpod
# Refactor cmd `tasks` and `ports` supervisor related func

## Description

Currently command `gp tasks` and `gp ports` not handle errors properly, i.e.
- Command not stop when supervisor grpc connect failed (`supervisor.Dial` func)
- Log with Error not Fatal and without return
- Package of client of supervisor called `supervisor` can be confusing

This PR 
- Move command files into `package cmd`
- Rename truly supervisor package from `api` to `supervisor`
- Move client of supervisor into `supervisor-helper` folder
- Fix i.e. problem above

For `gp tasks` 

- We don't change its logic, in case break something. If we want to refactor it, another PR will be good
- Update `GetTasksList` func for the same [reason](https://github.com/gitpod-io/gitpod/pull/10388#discussion_r886344771)

## Related Issue(s)

Fixes #

## How to test

- Start a workspace with current PR in preview env
- cd /workspace/gitpod/components/gitpod-cli
- Check if `go run main.go tasks xxx` behave the same with `gp tasks xxx`
- Check if `gp ports list` behave the same with prev PR https://github.com/gitpod-io/gitpod/pull/10388

## Release Notes

```release-note
NONE
```

## Documentation
